src/deluge/model/mod_controllable/mod_controllable_audio.cpp

```
@@ -1621,197 +1629,332 @@ ModelStackWithThreeMainThings* ModControllableAudio::addNoteRowIndexAndStuff(Mod
-					newKnobPos = knobPos + offset;
-					newKnobPos = std::max(newKnobPos, lowerLimit);
-					newKnobPos = std::min(newKnobPos, 64_i32);
-					if (newKnobPos == knobPos) {
-						continue;
-					}
-				}
-				else {
-					if (midiEngine.midiTakeover == MIDITakeoverMode::JUMP) { //Midi Takeover Mode = Jump
-						newKnobPos = 64;
-						if (value < 127) {
-							newKnobPos = (int32_t)value - 64;
+						int32_t previousValue =
+						    modelStackWithParam->autoParam->getValuePossiblyAtPos(modPos, modelStackWithParam);
+						int32_t knobPos = modelStackWithParam->paramCollection->paramValueToKnobPos(
+						    previousValue, modelStackWithParam);
+						int32_t lowerLimit = std::min(-64_i32, knobPos);
+						newKnobPos = knobPos + offset;
+						newKnobPos = std::max(newKnobPos, lowerLimit);
+						newKnobPos = std::min(newKnobPos, 64_i32);
+						if (newKnobPos == knobPos) {
+							continue;
 						}
-						knob->previousPositionSaved = false;
 					}
-					else { //Midi Takeover Mode = Pickup or Value Scaling
-						/*
+					else {
+						newKnobPos = calculateKnobPosForMidiTakeover(modelStackWithParam, modPos, value, knob);
+					}
 
-						Step #1: Convert Midi Controller's CC Value to Deluge Knob Position Value
+					//Convert the New Knob Position to a Parameter Value
+					int32_t newValue =
+					    modelStackWithParam->paramCollection->knobPosToParamValue(newKnobPos, modelStackWithParam);
 
-						- Midi CC Values for non endless encoders typically go from 0 to 127
-						- Deluge Knob Position Value goes from -64 to 64
+					//Set the new Parameter Value for the MIDI Learned Parameter
+					modelStackWithParam->autoParam->setValuePossiblyForRegion(newValue, modelStackWithParam, modPos,
+					                                                          modLength);
 
-						To convert Midi CC Value to Deluge Knob Position Value, subtract 64 from the Midi CC Value
+					//	if (midiEngine.midiFollowDisplayParam) {
```

To do add a default to the menu hide/display mod encoder popups